src/components/views/DownloadModModal.vue

```diff
@@ -117,26 +118,34 @@ import ProfileModList from '../../r2mm/mods/ProfileModList';
             );
 
             this.$store.commit('download/setIsModProgressModalOpen', true);
-            setTimeout(() => {
-                ThunderstoreDownloaderProvider.instance.download(this.profile.asImmutableProfile(), tsMod, tsVersion, this.ignoreCache, (progress: number, modName: string, status: number, err: R2Error | null) => {
-                    try {
-                        if (status === StatusEnum.FAILURE) {
-                            this.$store.commit('download/setIsModProgressModalOpen', false);
-                            this.$store.commit('download/updateDownload', {assignId, failed: true});
-                            if (err !== null) {
-                                DownloadMixin.addSolutionsToError(err);
-                                throw err;
+
+            const tsCombo = new ThunderstoreCombo();
+            tsCombo.setMod(tsMod);
+            tsCombo.setVersion(tsVersion);
+
+            setTimeout(async () => {
+                try {
+                    const downloadedMods = await ThunderstoreDownloaderProvider.instance.download(this.profile.asImmutableProfile(), tsCombo, this.ignoreCache, (progress: number, modName: string, status: number, err: R2Error | null) => {
+                        try {
+                            if (status === StatusEnum.FAILURE) {
+                                this.$store.commit('download/setIsModProgressModalOpen', false);
+                                this.$store.commit('download/updateDownload', {assignId, failed: true});
+                                if (err !== null) {
+                                    DownloadMixin.addSolutionsToError(err);
+                                    throw err;
+                                }
+                            } else if (status === StatusEnum.PENDING) {
+                                this.$store.commit('download/updateDownload', {assignId, progress, modName});
                             }
-                        } else if (status === StatusEnum.PENDING) {
-                            this.$store.commit('download/updateDownload', {assignId, progress, modName});
+                        } catch (e) {
+                            this.$store.commit('error/handleError', R2Error.fromThrownValue(e));
                         }
-                    } catch (e) {
-                        this.$store.commit('error/handleError', R2Error.fromThrownValue(e));
-                    }
-                }, async (downloadedMods) => {
+                    });
                     await this.downloadCompletedCallback(downloadedMods);
                     this.$store.commit('download/setIsModProgressModalOpen', false);
-                });
+                } catch (e) {
+                    this.$store.commit('error/handleError', R2Error.fromThrownValue(e));
```

Comment:
Should this close the progress modal before opening the error modal, and/or do some other cleanup?

Reply:
Yes, closing the modal makes sense.
_Maybe_ it would be wise to move this modal open state tracking to the `ModalsModule` and close all modals (or at least the progress modal) when an error modal is opened...

Reply:
We should probably `return` from the `catch` to avoid the rest of the code getting executed after failure.